File: 485_MaxConsecutiveOnes.cpp
//46.49%
class Solution {
public:
   int findMaxConsecutiveOnes(vector<int>& nums) {
        int currCnt(0), maxCnt(0);
		int zeroFlag(true);
		for(auto num : nums){
			if(num == 1){
				if(zeroFlag){
					zeroFlag = false;
					currCnt = 1;
				}
				else
					currCnt++;
			}else{
				if(!zeroFlag){
					if(currCnt > maxCnt)
						maxCnt = currCnt;
					zeroFlag = true;
				}
			}
		}
		return max(maxCnt, currCnt);
   }
};

File: 485_MaxConsecutiveOnes.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        currCnt, maxCnt, zeroFlag = 0, 0, True
        for num in nums:
            if num == 1:
                if zeroFlag:
                    currCnt = 1
                    zeroFlag = False
                else:
                    currCnt += 1
            else:
                if not zeroFlag:
                    zeroFlag = True
                    maxCnt = max(maxCnt, currCnt)
        return max(maxCnt, currCnt)